network/hits.go

```diff
@@ -0,0 +1,101 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package network
+
+import (
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/graph"
+)
+
+// HubAuthority is a Hyperlink-Induced Topic Search hub-authority score pair.
+type HubAuthority struct {
+	Hub       float64
+	Authority float64
+}
+
+// HITS returns the Hyperlink-Induced Topic Search hub-authority scores for
+// nodes of the directed graph g. HITS terminates when the 2-norm of the
+// vector difference between iterations is below eps. The returned map is
+// keyed on the graph node IDs.
+func HITS(g graph.DirectedGraph, eps float64) map[int]HubAuthority {
+	nodes := g.NodeList()
+
+	// Make a topological copy of g with dense node IDs.
+	indexOf := make(map[int]int, len(nodes))
+	for i, n := range nodes {
+		indexOf[n.ID()] = i
+	}
+	to := make([][]int, len(nodes))
```

Comment:
It could also be "range over the nodes v links to". For example, this latter meaning is used in PageRank where "to" contains the successors of the node. Maybe into isn't any better, but there must be some way to make reading easier.